core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -528,7 +528,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         new StageUIData
       })
       val taskData = stageData.taskData.get(taskId)
-      val metrics = TaskMetrics.fromAccumulatorInfos(accumUpdates)
+      val accumsFiltered = if (conf.get(TASK_METRICS_TRACK_UPDATED_BLOCK_STATUSES)) {
+        accumUpdates
+      } else {
+        accumUpdates.filter(info => info.name.isDefined && info.update.isDefined && info.name !=
```

Comment:
to be more clear, I think we should just do an assert here to make sure there is not UPDATED_BLOCK_STATUSES accumulator updates, instead of doing a filter.